generators/base-generator.js

```diff
@@ -117,7 +117,9 @@ module.exports = yeoman.Base.extend({
             return prompt.commonFilter.call(this, input);
           }.bind(this);
         }
-        if (!prompt.hasOwnProperty('validate') && prompt.hasOwnProperty('name')) {
+        if (prompt.hasOwnProperty('validate')) {
+          newPrompt.validate = newPrompt.validate.bind(this);
```

Comment:
Yeah, I'm not sure we should do this, unless we go through and add the binding to all other user supplied prompt functions. My inclination is to not do this and to do it in the prompt definition. In the case of the generated functions we have to perform this binding because the user essentially doesn't have a way to control that. ... or maybe they do... hmm... I should probably think on this some more... :)